tasks/node_users.yml

```diff
@@ -2,15 +2,53 @@
 # It is important that the dbadmin users on each node in a cluster can have passwordless ssh access to the other machines
 # This is done with a shared private ssh key
 
-# todo first create the dbadmin user and the dbadmin_group
+- name: Create dbadmin group
+  group: name="{{vertica_dbadmin_group}}" state=present
 
-# todo challenges are doing a shared private key that is secure, we can generate one and put it in a variable. Subsequent machines
-# will check the variable and their setup. On the next run the first machine will see it has one set and put it into the variable.
-# to generate a command such as "ssh-keygen -t rsa -b 2048 -f {{vertica_dbadmin_home}}/.ssh/id_rsa -q -N ''"
-# Add the public key to .ssh/authorized_keys
+- name: Create dbadmin user
+  user: name="{{vertica_dbadmin_user}}" home="{{vertica_dbadmin_home}}" group="{{vertica_dbadmin_group}}" shell=/bin/bash
 
-# todo create ~dbadmin/.ssh as the dbadmin user/group mode 700
-# Add in a .ssh/config with content="StrictHostKeyChecking no"
+- name: Create ssh directory in vertica home directory
+  file: path="{{vertica_dbadmin_home}}"/.ssh owner="{{vertica_dbadmin_user}}" group="{{vertica_dbadmin_group}}" mode=0700 state=directory
+
+- name: Create config for ssh 
+  copy: dest={{vertica_dbadmin_home}}/.ssh/config owner="{{vertica_dbadmin_user}}" group="{{vertica_dbadmin_group}}" mode=0600 content="StrictHostKeyChecking no" 
+
+- name: Check status abadmin ssh file
+  stat: path={{vertica_ssh_key_location}}/id_rsa
+  register: status 
+  when: vertica_ssh_private_key is not defined
+
+- name: Create ssh keys
+  command: ssh-keygen -t rsa -b 2048 -f {{vertica_ssh_key_location}}/id_rsa -q -N ''
+  when: not status.stat.exists and vertica_ssh_private_key is not defined
+
+- name: Read in private ssh key
+  slurp: src={{vertica_ssh_key_location}}/id_rsa
+  register: private_key_info
+  when: vertica_ssh_private_key is not defined
+
+- name: Read in public ssh key
+  slurp: src={{vertica_ssh_key_location}}/id_rsa.pub
+  register: public_key_info
+  when: vertica_ssh_public_key is not defined
+
+- name: Set public and private key variables
+  set_fact: vertica_ssh_private_key="{{private_key_info.content | b64decode}}" vertica_ssh_public_key="{{public_key_info.content | b64decode}}"
+  when: vertica_ssh_private_key is not defined
+
+- name: Create id_rsa
+  copy: dest={{vertica_dbadmin_home}}/.ssh/id_rsa owner="{{vertica_dbadmin_user}}" group="{{vertica_dbadmin_group}}" mode=0600 content="{{vertica_ssh_private_key}}" 
+
+- name: Create id_rsa.pub
+  copy: dest={{vertica_dbadmin_home}}/.ssh/id_rsa.pub owner="{{vertica_dbadmin_user}}" group="{{vertica_dbadmin_group}}" mode=0600 content="{{vertica_ssh_public_key}}"
+
+- name: Add public key to authorized keys
+  authorized_key: user="{{vertica_dbadmin_user}}" key="{{vertica_ssh_public_key}}"
+
+- name: Create user directory under vertica for dbadmin
```

Comment:
The ssh section is great I think we ended up with a nice solution there. Since the ssh section is so long it is worth considering moving this section above it so the file is roughly organized into the non-ssh stuff and the ssh stuff.